systems/trajectory_optimization/multiple_shooting.h

```diff
@@ -282,31 +282,50 @@ class MultipleShooting {
     DoAddRunningCost(g(0, 0));
   }
 
+  /// Adds a constraint to all breakpoints, where any instances in `vars` of
+  /// time(), state(), and/or input() placeholder variables, as well as
+  /// placeholder variables returned by calls to NewSequentialVariable(), are
+  /// substituted with the relevant variables for each time index.
+  /// @return A vector of the bindings added to each knot point.
+  /// @pydrake_mkdoc_identifier{shared_ptr}
+  template <typename C>
+  std::vector<solvers::Binding<C>> AddConstraintToAllKnotPoints(
+      std::shared_ptr<C> constraint,
+      const Eigen::Ref<const VectorX<symbolic::Variable>>& vars) {
+    std::vector<solvers::Binding<C>> bindings;
+    for (int i = 0; i < N_; i++) {
+      bindings.push_back(prog_.AddConstraint(
+          constraint, sequential_expression_manager_.GetVariables(vars, i)));
+    }
+    return bindings;
+  }
+
   /// Adds a constraint to all breakpoints, where any instances of time(),
   /// state(), and/or input() placeholder variables, as well as placeholder
   /// variables returned by calls to NewSequentialVariable(), are substituted
   /// with the relevant variables for each time index.
-  /// @return A vector of the constraints added to each knot point.
+  /// @return A vector of the bindings added to each knot point.
```

Comment:
Should this change entail renaming the `constraints` variable below for consistency?